ddtrace/profiling/collector/_task.pyx

```diff
@@ -1,25 +1,35 @@
+import sys
 import weakref
 
 from ddtrace.internal import compat
-from ddtrace.internal import nogevent
+from ddtrace.vendor.wrapt.importer import when_imported
 
 from .. import _asyncio
 from .. import _threading
 
 
-try:
-    import gevent.hub
-    import gevent.thread
-    from greenlet import getcurrent
-    from greenlet import greenlet
-    from greenlet import settrace
-except ImportError:
-    _gevent_tracer = None
-else:
+_gevent_tracer = None
+
+
+@when_imported("gevent")
+def install_greenlet_tracer(gevent):
+    global _gevent_tracer
+
+    try:
+        import gevent.hub
+        import gevent.thread
+        from greenlet import getcurrent
+        from greenlet import greenlet
+        from greenlet import settrace
+    except ImportError:
+        # We don't seem to have the required dependencies.
+        return 
 
     class DDGreenletTracer(object):
-        def __init__(self):
+        def __init__(self, gevent):
```

Comment:
What's the significance of attaching the gevent import to the `DDGreenletTracer` object?

Reply:
This way we get the reference to the module that triggered the import and we don't have to go look it up from `sys.modules`.